floodviz/maputils.py

```diff
@@ -0,0 +1,51 @@
+import requests
+
+# function to generate a dictionary of site data after querying NWIS
+def site_dict(site_list, url_prefix):
+    # generate the string of site ids for the url
+    id_input_string = ",".join(site_list)
+
+    # create the url
+    url = url_prefix + "/?format=rdb&sites=" + id_input_string + "&siteStatus=all"
+
+    # get data from url
+    data = ""
+    req = requests.get(url)
+    for line in req.text.splitlines():
+        if not line.startswith("#"):
+            data += line + '\n'
+
+    # create array by splitting lines
+    data = data.split('\n')
+
+    # make a list of dicts  from data
+    fields = data[0].split('\t')
+    dnice = []
+    for line in data[2:]:
+        line = line.split('\t')
+        line_dict = dict(zip(fields, line))
+        dnice.append(line_dict)
+
+    # remove bad element from end of list
+    dnice = dnice[:-1]
+    return dnice
+
+
+# function to write data from NWIS to a geojson file
+def write_geojson(filename, data):
+    with open(filename, "w") as f:
+        f.write("{ \"type\": \"FeatureCollection\", \"features\": [ \n")
+
+        count = -1
+        for datum in data:
```

Comment:
Consider using enumerate rather than incrementing a counter. Garritt used it in his work for the hydrograph. It was new to me too ... why I love to do code reviews because the reviewer learns things to!